howto/writing_feature_files.md

```diff
@@ -0,0 +1,147 @@
+# Writing feature files
+
+<!-- index {{{ -->
+* [Introduction](#introduction)
+* [Summary](#summary)
+* [The `Given` step](#the_given_step)
+* [The `When` step](#the_when_step)
+* [The `Then` step](#the_then_step)
+<!-- }}} -->
+
+<!-- Introduction {{{ -->
+## Introduction
+<a name="introduction" href=""></a>
+
+As feature files are written using the English language, we need some rules to
+standarize how we write these files. This document will serve the purpose of
+being a reference to which developers can come back or point to.
+<!-- }}} -->
+
+<!-- Summary {{{ -->
+<a name="summary" href=""></a>
+## Summary
+
+### Given
+
+* Describe the current state
+* Use past tense when describing an action that has happened
+* Use present tense when describing a given state
+* Always use the passive voice
+
+### When
+
+* Describe the actions of the scenario
+* Always use present tense
+* Always use the active voice
+
+### Then
+
+* Describe what should happen
+* Always use present tense
+* Always use affirmative voice
+* Always use the passive voice
+<!-- }}} -->
+
+<!-- Given {{{ -->
+<a name="the_given_step" href=""></a>
+## The `Given` step
+
+### Describing the current state
+
+These steps define the current state, they do not describe what needs to be
+done to get there:
+
+```feature
+# wrong
+Given the user logged in
+
+# right
+Given the user is logged in
+```
+
+The difference in this example is sublte, which is intended.
+In the wrong example an action is performed and no concrete state was
+mentioned. It's unclear what the resulting state will be after the user has
+logged in.
+
+The right example on the other hand describes the current sitution, regardless
+of what happened to get there. The user could've logged in himself, he could've
+been logged in automatically as his session is still valid, he could've used
+oAuth2 or a simple basic auth. Regardless of how the user has been logged in,
+the state is clear: The user is logged in.
```

Comment:
I would suggest adding more arguments on why the wrong example is wrong. Some could argue that `Given the user logged in` doesn't express the means of logging in either, just that the user is logged in (state).

Reply:
@vilkg I've added a bit more to make it a bit more clear, did that help or is it still unclear?

Reply:
Yeah, second example is much better @Mohammer5 !